src/main/java/by/custom/utilcalculator/domain/UserProgress.java

```diff
@@ -12,17 +14,23 @@ public class UserProgress implements Serializable {
     private VolumeOfEngine volumeOfEngine = null;
     private final String chatID;
     private Step currentQuestion;
+    private Enum[] stepsData;
+    private PassengerStepsChecker passengerStepsChecker;
```

Comment:
This field is redundant.  UserProgress should store only user data. Look POJO object https://naveen-metta.medium.com/unleashing-the-power-of-plain-old-java-objects-pojos-in-modern-development-19cb57638f6

Reply:
The article was wonderful. The main thought is "use POJO for data-classes". That's why we don't need Enum[] and StepsChecker here